### PR TITLE
[v2] add FromEnv suffix to metadata resolved from env

### LINKED ISSUE #1044: Renaming metadata propertied to indicate source of the value
Today in the `metadata` section of each scaler, there are some properties that are intended to be resolved from the scaleTarget of the `ScaledObject`. These are mostly meant to refer to the secrets needed (if any) for each scaler to access the event source.

For example `aws-sqs-queue` looks something like this 

```yaml
...
    type: aws-sqs-queue
    metadata:
      awsAccessKeyID: AWS_ACCESS_KEY_ID
      awsSecretAccessKey: AWS_SECRET_ACCESS_KEY
      
```

This means that the scaleTarget (e.g: Deployment) will have a container that has those defined as environment variables (whether directly or sourced from a `v1.Secret` or a `v1.ConfigMap`). KEDA resolves those values and passes the actual value of the key/secret in this example to the `aws-sqs-queue` scaler. 

Based on feedback and some discussion on the community meeting, we are considering renaming all metadata keys that behave in that way with a standard suffix, for example `*FromEnv`

This is a look at all such settings for all scalers. 

| Scaler | values | update |
|--------|--------|--------|
| `azure-blob` | `connection` (**Default**: `AzureWebJobsStorage`) | `connectionFromEnv` |
| `azure-monitor` | `activeDirectoryClientId` <br /> `activeDirectoryClientPassword` | `activeDirectoryClientIdFromEnv` <br /> `activeDirectoryClientPasswordFromEnv` |
| `azure-queue` | `connection` (**Default**: AzureWebJobsStorage) | `connectionFromEnv` |
| `azure-servicebus` | `connection` | `connectionFromEnv` |
| `azure-eventhub` | `storageConnection` (**Default**: `AzureWebJobsStorage`) <br /> `connection` (**Default**: `EventHub`) | `storageConnectionFromEnv` <br /> `connectionFromEnv` |
| `aws-sqs-queue` <br /> `aws-cloudwatch` <br /> `aws-kinesis-stream` | `awsAccessKeyID` (**Default**: `AWS_ACCESS_KEY_ID`) <br /> `awsSecretAccessKey` (**Default**: `AWS_SECRET_ACCESS_KEY`) | `awsAccessKeyIDFromEnv` <br /> `awsSecretAccessKeyFromEnv` |
| `kafka` | _(none)_ | _(none)_
| `rabbitmq` | `apiHost` <br /> `host` | `apiHostFromEnv` <br /> `host` |
| `prometheus` | _(none)_ | _(none)_ |
| `cron` | _(none)_ | _(none)_ |
| `redis` <br /> `redis-streams` | `address` <br /> `host` <br /> `port` <br /> `password` | `addressFromEnv` <br /> `hostFromEnv` <br /> ~~`port`~~ <br /> `passwordFromEnv`
| `gcp-pubsub` | `credentials` | `credentialsFromEnv` |
| `external` | _(any matching value)_ | _(any matching value with `FromEnv` suffix)_
| `liiklus` | _(none)_ | _(none)_ |
| `stan` | _(none)_ | _(none)_ |
| `huawei-cloudeye` | | _(none)_ | _(none)_ |
| `postgresql` | `connection` <br /> `password` | `connectionFromEnv` <br /> `passwordFromEnv` |
| `mysql` | `connectionString` <br /> `password` | `connectionStringFromEnv` <br /> `passwordFromEnv` |

There is also probably a chance to make things a bit more consistent between the scalers as a popular ask from @tomkerkhove  :)

### PR DESCRIPTION
**Pull request by ahmelsayed**
Signed-off-by: Ahmed ElSayed <ahmels@microsoft.com>

| Scaler | values | update |
|--------|--------|--------|
| `azure-blob` | `connection` (**Default**: `AzureWebJobsStorage`) | `connectionFromEnv` |
| `azure-monitor` | `activeDirectoryClientId` <br /> `activeDirectoryClientPassword` | `activeDirectoryClientIdFromEnv` <br /> `activeDirectoryClientPasswordFromEnv` |
| `azure-queue` | `connection` (**Default**: AzureWebJobsStorage) | `connectionFromEnv` |
| `azure-servicebus` | `connection` | `connectionFromEnv` |
| `azure-eventhub` | `storageConnection` (**Default**: `AzureWebJobsStorage`) <br /> `connection` (**Default**: `EventHub`) | `storageConnectionFromEnv` <br /> `connectionFromEnv` |
| `aws-sqs-queue` <br /> `aws-cloudwatch` <br /> `aws-kinesis-stream` | `awsAccessKeyID` (**Default**: `AWS_ACCESS_KEY_ID`) <br /> `awsSecretAccessKey` (**Default**: `AWS_SECRET_ACCESS_KEY`) | `awsAccessKeyIDFromEnv` (**Default**: `AWS_ACCESS_KEY_ID`) <br /> `awsSecretAccessKeyFromEnv` (**Default**: `AWS_SECRET_ACCESS_KEY`) |
| `kafka` | _(none)_ | _(none)_
| `rabbitmq` | `apiHost` <br /> `host` | `apiHostFromEnv` <br /> `host` |
| `prometheus` | _(none)_ | _(none)_ |
| `cron` | _(none)_ | _(none)_ |
| `redis` <br /> `redis-streams` | `address` <br /> `host` <br /> `port` <br /> `password` | `addressFromEnv` <br /> `hostFromEnv` <br /> `portFromEnv` <br /> `passwordFromEnv`
| `gcp-pubsub` | `credentials` | `credentialsFromEnv` |
| `external` | _(any matching value)_ | _(any matching value with `FromEnv` suffix)_
| `liiklus` | _(none)_ | _(none)_ |
| `stan` | _(none)_ | _(none)_ |
| `huawei-cloudeye` | _(none)_ | _(none)_ |
| `postgresql` | `connection` <br /> `password` | `connectionFromEnv` <br /> `passwordFromEnv` |
| `mysql` | `connectionString` <br /> `password` | `connectionStringFromEnv` <br /> `passwordFromEnv` |

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Tests have been added
- [x] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs
- [ ] Changelog has been updated

Fixes #1044

### CODE REVIEW

File: pkg/scalers/azure_eventhub_scaler.go [hunk 1]
```diff
@@ -20,13 +20,11 @@ import (
...
-	thresholdMetricName              = "unprocessedEventThreshold"
-	defaultEventHubConsumerGroup     = "$Default"
-	defaultEventHubConnectionSetting = "EventHub"
-	defaultStorageConnectionSetting  = "AzureWebJobsStorage"
-	defaultBlobContainer             = ""
+	defaultEventHubMessageThreshold = 64
+	eventHubMetricType              = "External"
+	thresholdMetricName             = "unprocessedEventThreshold"
+	defaultEventHubConsumerGroup    = "$Default"
+	defaultBlobContainer = ""
```
Comment (turbaszek) [inline pkg/scalers/azure_eventhub_scaler.go:27]:
```suggestion
	defaultEventHubConsumerGroup    = "$Default"
	defaultBlobContainer                         = ""
```

**Review by zroubalik — APPROVED** [review]
LGTM, great job!

**Comment by tomkerkhove** [comment]
Thanks a ton @ahmelsayed!

**Comment by zroubalik** [comment]
@ahmelsayed seems like there are multiple e2e test failures, seems like caused by this PR
https://github.com/kedacore/keda/runs/1069431785

**Review by tomkerkhove — COMMENTED** [review]
Just some follow-up questions

**Pull request merged**